framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -76,37 +76,45 @@
  *       error if any are found.
  * </ol>
  *
- * <p>Steps 3 and 4 are separated so that an error is issued only once per invalid expression string
+ * <p>Steps 2 and 3 are separated so that an error is issued only once per invalid expression string
  * rather than every time the expression string is parsed. (The expression string is parsed multiple
  * times because annotated types are created multiple times.)
  */
 public class DependentTypesHelper {
+
+    /** AnnotatedTypeFactory */
     protected final AnnotatedTypeFactory factory;
 
     /** A map of annotation classes to the names of their elements that are Java expressions. */
-    private Map<Class<? extends Annotation>, List<String>> annoToElements;
+    private final Map<Class<? extends Annotation>, List<String>> annoToElements;
 
     /**
-     * The {@code ExpressionErrorChecker} that scans an annotated type and returns a list of {@link
-     * DependentTypesError}.
+     * The {@code ExpressionErrorCollector} that scans an annotated type and returns a list of
+     * {@link DependentTypesError}.
      */
-    private final ExpressionErrorChecker expressionErrorChecker;
+    private final ExpressionErrorCollector expressionErrorCollector =
+            new ExpressionErrorCollector();
 
     /**
-     * A scanner that standardizes Java expression strings in dependent type annotations. Call
-     * {@code StandardizeTypeAnnotator#init(JavaExpressionContext, TreePath, boolean, boolean)}
-     * before each use.
+     * A scanner that applies a function to each {@link AnnotationMirror} in a given {@code
```

Comment:
Please make explicit whether this returns a new one or side-effects an existing one.